### PR TITLE
fix(gateway): add node status for some list APIs

### PR DESCRIPTION
**Pull request by lafirest**

### CODE REVIEW

File: apps/emqx_gateway/src/emqx_gateway_api_listeners.erl [hunk 1]
```diff
@@ -80,7 +83,39 @@ paths() ->
 
 listeners(get, #{bindings := #{name := Name0}}) ->
     with_gateway(Name0, fun(GwName, _) ->
-        {200, emqx_gateway_conf:listeners(GwName)}
+        Listeners = emqx_gateway_conf:listeners(GwName),
+        ListenOns = lists:map(
+            fun(#{id := Id, <<"bind">> := BinListenOn}) ->
+                {Id, erlang:binary_to_integer(BinListenOn)}
```
Comment (HJianBo) [inline apps/emqx_gateway/src/emqx_gateway_api_listeners.erl:89]: Is there have a crash error due to the `BinListenOn` is a `integer()` or `ip_port()` type?

Comment (HJianBo) [reply apps/emqx_gateway/src/emqx_gateway_api_listeners.erl:89]: nit: Maybe we can move the L86-L118 into emqx_gateway_http module to get more short lines for here🤔?

File: apps/emqx_gateway/src/emqx_gateway_api_listeners.erl [hunk 2]
```diff
@@ -553,14 +588,50 @@ params_paging_in_qs() ->
 roots() ->
     [listener].
 
+fields(listener_node_status) ->
+    [
+        {current_connections, mk(non_neg_integer(), #{desc => ?DESC(current_connections)})},
```
Comment (HJianBo) [inline apps/emqx_gateway/src/emqx_gateway_api_listeners.erl:593]: miss `max_connections` field

Comment (lafirest) [reply apps/emqx_gateway/src/emqx_gateway_api_listeners.erl:593]: The 'max_connections'  already exist in emqx_gateway_api:fields(xxx_listeners)

**Review by HJianBo — APPROVED** [review]

**Pull request merged**